lib/sqlitex/statement.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Sqlitex.Statement do
+  @moduledoc """
+  Provides an interface for working with sqlite prepared statements.
+
+  ```
+  iex(2)> {:ok, db} = Sqlitex.open(":memory:")
+  iex(3)> Sqlitex.query(db, "CREATE TABLE data (id, name);")
+  []
+  iex(6)> {:ok, statement} = Sqlitex.Statement.prepare(db, "INSERT INTO data VALUES (?, ?);")
+  iex(7)> Sqlitex.Statement.bind_values(statement, [1, "hello"])
+  iex(8)> Sqlitex.Statement.fetch_all(statement)
```

Comment:
I did think about this while writing the library, but wasn't sure what you'd want.  Guess I didn't see the parallels with the `Sqlite.exec` - will add it in.